src/gsi/cplr_wrwrfmassa.f90

```diff
@@ -2200,18 +2261,158 @@ subroutine wrwrfmassa_netcdf_wrf(this,mype)
              	
   ! for hydrometeors      
              if(l_hydrometeor_bkio .and. n_actual_clouds>0) then
-                all_loc(j,i,kqc)=ges_qc(j,i,k)
-                all_loc(j,i,kqi)=ges_qi(j,i,k)
-                all_loc(j,i,kqr)=ges_qr(j,i,k)
-                all_loc(j,i,kqs)=ges_qs(j,i,k)
-                all_loc(j,i,kqg)=ges_qg(j,i,k)
-                all_loc(j,i,kqnr)=ges_qnr(j,i,k)
-                all_loc(j,i,kqni)=ges_qni(j,i,k)
-                all_loc(j,i,kqnc)=ges_qnc(j,i,k)
-                if(dbz_exist.and.if_model_dbz)all_loc(j,i,kdbz)=ges_dbz(j,i,k)
-                all_loc(j,i,ktt)=ges_tten(j,i,k,it)
-             endif
-  
+! --- direct reflectivity DA---> run CV transform on hydrometeor variables
+                if ( l_use_dbz_directDA) then
+                   all_loc(j,i,kqc)=ges_qc(j,i,k)
+                   all_loc(j,i,kqi)=ges_qi(j,i,k)
+                   all_loc(j,i,ktt)=ges_tten(j,i,k,it)
+
+!                set up the tiniy values for qr/qs/qg and convert log-transformed variable back to normal vaiable
+                   qr_tmp=zero
+                   qs_tmp=zero
+                   qg_tmp=zero
+
+                   if ( l_use_cvpqx ) then
+                      if ( l_use_cvpqx_pval .eq. 0. ) then ! CVlogq
+                         if (ges_tsen(j,i,k,it) .gt. 274.15) then
+                            qr_min=2.9E-6_r_kind
+                            qr_thrshd=qr_min * one_tenth
+                            qs_min=0.1E-9_r_kind
+                            qs_thrshd=qs_min
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         else if (ges_tsen(j,i,k,it) .le. 274.15 .and. ges_tsen(j,i,k,it) .ge. 272.15) then
+                            qr_min=2.0E-6_r_kind
+                            qr_thrshd=qr_min * one_tenth
+                            qs_min=1.3E-7_r_kind
+                            qs_thrshd=qs_min * one_tenth
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         else if (ges_tsen(j,i,k,it) .lt. 272.15) then
+                            qr_min=0.1E-9_r_kind
+                            qr_thrshd=qr_min
+                            qs_min=6.3E-6_r_kind
+                            qs_thrshd=qs_min * one_tenth
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         end if
+!                        convert log(qr/qs/qg) back to qr/qs/qg
+                         if (mype==0 .AND. i==3 .AND. j==3 .AND. k==3) then
```

Comment:
convert this kind of write to comments or move them out of the loop.

Reply:
I modified the code according to your comments. I took out these codes from loops and make them print one time at the start of the routine using if(mype==0).

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-83013677e0fb286b5f115ca5766f4b67266ed22b3d70b8867ed865db67ab8f50R3213-R3226